python/Ganga/Utility/Runtime.py

```diff
@@ -257,12 +257,14 @@ def loadPlugins(environment):
             logger.error('problems with loading Named Templates for %s', n)
             logger.error('Reason: %s' % str(err))
 
-    for r in allRuntimes.values():
+    for n, r in zip(allRuntimes.keys(), allRuntimes.values()):
         try:
             r.loadPlugins()
         except Exception as err:
-            logger.error('problems with loading Named Templates for %s', n)
+            from Ganga.Core.exceptions import GangaException
+            logger.error('problems with loading Plugin %s', n)
             logger.error('Reason: %s' % str(err))
+            raise GangaException("Failed to load plugin: %s. Ganga will now shutdown to prevent job corruption." % n)
```

Comment:
I've fixed this to what is requested as per the latest commit but this whole file (in fact a _lot_ of files) need to have the import situation sorted out.
Fixing 1 or 2 methods ad-hoc just makes the files messy imo and instead of each method importing what it needs where it needs it (yes a bad thing) we now have a horrible mix of global and local imports which should probably be 90%+ global imports due to a handful of circular dependencies.